torchtune/datasets/_preference.py

```diff
@@ -10,64 +10,98 @@
 from datasets import load_dataset
 from torch.utils.data import Dataset
 
-from torchtune.data import CROSS_ENTROPY_IGNORE_IDX, InstructTemplate, Message
+from torchtune.data import CROSS_ENTROPY_IGNORE_IDX, PromptTemplate
 
 from torchtune.modules.tokenizers import ModelTokenizer
+from torchtune.modules.transforms import Transform
 
 
 class PreferenceDataset(Dataset):
     """
-    Class that supports any custom dataset with instruction-based prompts and a
-    configurable template.
+    Primary class for fine-tuning via RLHF/PPO or DPO methods on a preference dataset
```

Comment:
Is this true for PPO? I thought there we use text completion dataset

Reply:
Should be:

fine-tuning for preference modelling techniques (e.g. training a preference model for RLHF, or directly optimizing a model through DPO).

Reply:
sorry, didn't realize PPO is text completion